### PR TITLE
Atcommand setcard

### PR DESCRIPTION
**Pull request by Atemo**
* **Addressed Issue(s)**: -

* **Server Mode**: -

* **Description of Pull Request**: 

Suggestion of an utility GM command.
This allows to add/remove a card in the given slot (0-3) on equipped equipment.

### CODE REVIEW

File: doc/atcommands.txt [hunk 1]
```diff
@@ -1201,6 +1201,27 @@ Opens the enchantgrade UI.
...
+8: Left Accessory
+16: Body Armor
+32: Left Hand
+64: Shoes
+128: Right Accessory
+256: Top Headgear
+512: Mid Headgear
+
+<slot> number: 0-3.
+<card> can actually be any existing item. Put 0 to remove the card.
```
Comment (Lemongrass3110) [inline doc/atcommands.txt:1221]:
```suggestion
<card> can actually be any existing card or enchant item. Put 0 to remove it.
```

File: src/map/atcommand.cpp [hunk 2]
```diff
@@ -10809,6 +10809,65 @@ ACMD_FUNC( roulette ){
...
+		sprintf(atcmd_output, msg_txt(sd,1005), EQP_HEAD_TOP); // %d: Top Headgear
+		clif_displaymessage(fd, atcmd_output);
+		sprintf(atcmd_output, msg_txt(sd,1006), EQP_HEAD_MID); // %d: Mid Headgear
+		return -1;
+	}
+
+	if (card_id != 0) {
+		std::shared_ptr<item_data> item_data = item_db.find( card_id );
+
+		if (item_data == nullptr)
```
Comment (Lemongrass3110) [inline src/map/atcommand.cpp:10851]: A message to the user would be helpful.

File: src/map/atcommand.cpp [hunk 3]
```diff
@@ -10809,6 +10809,65 @@ ACMD_FUNC( roulette ){
...
+	int16 i = pc_checkequip(sd, position);
+
+	if (i < 0)
+		return -1;
+
+	int current_position = sd->inventory.u.items_inventory[i].equip;
+
+	sd->inventory.u.items_inventory[i].card[slot] = card_id;
+	pc_unequipitem(sd, i, 3);
+	clif_delitem(sd, i, 1, 3);
```
Comment (Lemongrass3110) [inline src/map/atcommand.cpp:10864]:
```suggestion
	clif_delitem(sd, i, 1, 0);
```

**Review by Lemongrass3110 — COMMENTED** [review]
Maybe add a small entry to atcommands.yml as well?

File: src/map/atcommand.cpp [hunk 4]
```diff
@@ -10809,6 +10809,65 @@ ACMD_FUNC( roulette ){
...
+		clif_displaymessage(fd, atcmd_output);
+		sprintf(atcmd_output, msg_txt(sd,1006), EQP_HEAD_MID); // %d: Mid Headgear
+		return -1;
+	}
+
+	if (card_id != 0) {
+		std::shared_ptr<item_data> item_data = item_db.find( card_id );
+
+		if (item_data == nullptr)
+			return -1;
```
[the remark this answers is not in the document]
Comment (Atemo) [reply src/map/atcommand.cpp:10852]: hmm before that, actually the other atcommands don't have many checks (GM commands are permissive). For example `@item2` doesnot check cards, `@refine` can refine any equipment - like costumes - etc..

Maybe we shouldn't worry about the checks (if the item is type card, if the card is applicable for that eq..) for this command as well?

File: src/map/atcommand.cpp [hunk 5]
```diff
@@ -10809,6 +10809,79 @@ ACMD_FUNC( roulette ){
...
+
+	if (i < 0) {
+		clif_displaymessage(fd, msg_txt(sd,1528)); // You are not wearing any equipment in this position.
+		return -1;
+	}
+
+	int current_position = sd->inventory.u.items_inventory[i].equip;
+
+	pc_unequipitem(sd, i, 3);
+	sd->inventory.u.items_inventory[i].card[slot] = card_id;
```
Comment (Lemongrass3110) [inline src/map/atcommand.cpp:10878]:
```suggestion
	log_pick_pc( sd, LOG_TYPE_COMMAND, -1, sd->inventory.u.items_inventory[i] );
	sd->inventory.u.items_inventory[i].card[slot] = card_id;
	log_pick_pc( sd, LOG_TYPE_COMMAND, 1, sd->inventory.u.items_inventory[i] );
```

**Review by Lemongrass3110 — APPROVED** [review]

File: src/map/atcommand.cpp [hunk 6]
```diff
@@ -10809,6 +10809,81 @@ ACMD_FUNC( roulette ){
...
+
+	if (i < 0) {
+		clif_displaymessage(fd, msg_txt(sd,1528)); // You are not wearing any equipment in this position.
+		return -1;
+	}
+
+	int current_position = sd->inventory.u.items_inventory[i].equip;
+
+	pc_unequipitem(sd, i, 3);
+	log_pick_pc( sd, LOG_TYPE_COMMAND, -1, sd->inventory.u.items_inventory[i] );
```
Comment (aleos89) [inline src/map/atcommand.cpp:10878]:
```suggestion
	log_pick_pc( sd, LOG_TYPE_COMMAND, -1, &sd->inventory.u.items_inventory[i] );
```

File: src/map/atcommand.cpp [hunk 7]
```diff
@@ -10809,6 +10809,81 @@ ACMD_FUNC( roulette ){
...
+	sd->inventory.u.items_inventory[i].card[slot] = card_id;
+	log_pick_pc( sd, LOG_TYPE_COMMAND, 1, sd->inventory.u.items_inventory[i] );
```
Comment (aleos89) [inline src/map/atcommand.cpp:10880]:
```suggestion
	log_pick_pc( sd, LOG_TYPE_COMMAND, 1, &sd->inventory.u.items_inventory[i] );
```

**Review by aleos89 — APPROVED** [review]

**Pull request merged**